drivers/flash/flash_sam0.c

```diff
@@ -74,16 +76,20 @@ static int flash_sam0_write_protection(const struct device *dev, bool enable);
 
 static inline void flash_sam0_sem_take(const struct device *dev)
 {
+#if defined(CONFIG_MULTITHREADING)
 	struct flash_sam0_data *ctx = dev->data;
```

Comment:
If we get "unused argument" warning here (we have this warning enabled if I remember correctly?), the preferred way to handle this should be to use `if (IS_ENABLED(CONFIG_MULTITHREADING)) { ... }` rather than adding `ARG_UNUSED` for `#else`.